### PR TITLE
Use Pyadjoint tagging for solves

### PR DESCRIPTION
**Pull request by jwallwork23**
In [Pyadjoint PR66](https://github.com/dolfin-adjoint/pyadjoint/pull/66) and [Firedrake PR2174](https://github.com/firedrakeproject/firedrake/pull/2174) I introduced optional tags for the Pyadjoint tape blocks associated with various high-level operations. Here I am applying them to the prognostic equation solves in Thetis, so that the blocks corresponding to these solves can be easily identified and extracted.

For example, this makes the extraction and plotting of adjoint solution fields associated with different equations straightforward.

### CODE REVIEW

File: thetis/rungekutta.py [hunk 1]
```diff
@@ -473,7 +473,7 @@ def __init__(self, equation, solution, fields, dt, options, bnd_conditions, term
         # Allocate tendency fields
         self.k = []
         for i in range(self.n_stages):
-            fname = '{:}_k{:}'.format(self.name, i)
+            fname = f'{self.name}_k{i}'
```
[the remark this answers is not in the document]
Comment (tkarna) [reply thetis/rungekutta.py:476]: Thanks we should use f strings where possible, most of these are from ages before such python features.

**Review by tkarna — COMMENTED** [review]
Thanks, this looks all right to me.

**Review by stephankramer — CHANGES_REQUESTED** [review]
Otherwise looks good, yeah

File: thetis/options.py [hunk 2]
```diff
@@ -14,6 +14,7 @@ class TimeStepperOptions(FrozenHasTraits):
     """Base class for all time stepper options"""
     name = 'Time stepper'
     solver_parameters = PETScSolverParameters({}).tag(config=True)
+    ad_block_tag = None
```
[the remark this answers is not in the document]
Comment (jwallwork23) [reply thetis/options.py:17]: Good point, thanks Stephan.

**Review by tkarna — APPROVED** [review]
Thanks, all good now!

**Pull request merged**